torch/autograd/function.py

```diff
@@ -48,6 +48,7 @@ def save_for_backward(self, *tensors: torch.Tensor):
         See :ref:`extending-autograd` for more details on how to use this method.
 
         Example::
+            >>> # xdoctest: +REQUIRES(env:TORCH_DOCTEST_AUTOGRAD)
```

Comment:
So how exactly do the REQUIRES annotations work, do we have to remember to put then on all the autograd related doctests?

Reply:
Short answer:

* The `+REQUIRES(env:...)` directive tests for an environment variable and skips the test if its not there.

* This is not required for new tests.

Long answer:

When xdoctests parses an example block, it breaks it into parts based on a few criteria like: "is there a got/want test?, or "is there a directive?". In the case of a directive, before it executes the code after it, it checks to see if the directive is both satisfied (based on the directive name and its arguments) and enabled (which is denoted by the `+` and nearly always true if you using directives, but technically you can have an `# xdoctest -REQUIRES(env:TORCH_DOCTEST_AUTOGRAD)`, which disables it. 

The takeaway is that if there is a `+REQUIRES(env:FOO)`, directive, that test will be skipped unless the `FOO` environment variable exists and is truthy in `os.environ`. (site node: you can also do this to test for the presense of dependencies, e.g. `+REQUIRES(module:torchvision)`).  

In `test/run_tests.py`, I have the `run_doctests` enable or disable these environment variables depending the environment the tests are running in. Currently `TORCH_DOCTEST_AUTOGRAD` is never enabled, likely because I found that the tests were causing issues. So having this environment variable serves as a quick way to enable / disable all of the associated tests, which can ideally allow them to be cleaned up and fixed in the future.

As for new autograd tests, you don't need to add this directive to them. If they run and they pass, that's great! However, if the reason I disabled them in the first place (I don't recall details, maybe I have a previous post that explains it) impacts the new test, then they will fail. This is a good thing, we want to know when doctests don't actually execute the way we thought they did, and we want to explicitly mark the iffy ones.